domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs

```diff
@@ -157,22 +157,24 @@ public class ContractsController : ControllerBase
             newEndDate,
             cancellationToken);
 
+        var contract = await service.GetById(id, meteringPointOwner, cancellationToken);
+
         return result switch
         {
             NonExistingContract => NotFound(),
             MeteringPointOwnerNoMatch => Forbid(),
             EndDateBeforeStartDate => ValidationProblem("EndDate must be after StartDate"),
-            SetEndDateResult.Success => await LogChangedEndDateAndReturnOk(activityLogEntryRepository, user, id),
+            SetEndDateResult.Success => await LogChangedEndDateAndReturnOk(activityLogEntryRepository, user, contract!.GSRN),
```

Comment:
You are going through the contract to get the gsrn? I thought you wanted to avoid that?

Also why is the null warning suppressed? Can't you make it null safe?

Reply:
Why would I avoid that? It's the only way to get the GSRN

I'm not suppressing any warnings, just telling the compiler I'm sure it's not null here since I jsut updated it

Reply:
ok